cupy/core/_kernel.pyx

```diff
@@ -1008,10 +961,105 @@ cpdef create_ufunc(name, ops, routine=None, preamble='', doc='',
             in_types, out_types = map(tuple, types)
         in_types = tuple([get_dtype(t).type for t in in_types])
         out_types = tuple([get_dtype(t).type for t in out_types])
-        _ops.append((in_types, out_types, rt))
+        return _Op(rt, in_types, out_types)
 
-    ret = ufunc(name, len(_ops[0][0]), len(_ops[0][1]), _ops, preamble,
-                loop_prep, doc, default_casting=default_casting)
-    return ret
+
+cdef class _Ops:
+    """A kernel routine representation with various dtype mappings.
+    """
+
+    cdef readonly tuple ops
+    cdef readonly int nin
+    cdef readonly int nout
+
+    def __init__(self, tuple ops):
+        assert len(ops) > 0
+        nin = ops[0].nin
+        nout = ops[0].nout
+        assert all(op.nin == nin for op in ops)
+        assert all(op.nout == nout for op in ops)
+        self.ops = ops
+        self.nin = nin
+        self.nout = nout
+
+    @staticmethod
+    cdef _Ops from_tuples(object ops, routine):
+        ops_ = []
+        for t in ops:
+            if isinstance(t, tuple):
+                typ, rt = t
```

Comment:
Some elements in `rt` can be `None`. Please add preprocess like as `rt = tuple([i or j for i, j in zip(rt, routine)])`.